File: src/types/estimator.ts
export type CityTier = 'metro' | 'tier1' | 'tier2' | 'tier3'
export type Quality = 'basic' | 'standard' | 'premium'
export type HouseType = 'independent' | 'villa' | 'row'
export type Floors = 1 | 2 | 3 | 4

export type EstimateInput = {
  plotSqft: number
  builtSqft: number
  floors: Floors
  houseType: HouseType
  cityTier: CityTier
  citySlug?: string
  quality: Quality
  basement?: boolean
  parking?: boolean
  lift?: boolean
  vastu?: boolean
  solarReady?: boolean
}

export type EstimateBreakdownItem = {
  key: string
  label: string
  amount: number
  percentage: number
  color: string
}

export type EstimateResult = {
  totalCost: number
  low: number
  high: number
  baseCost: number
  addonCost: number
  costPerSqft: number
  timelineMonths: number
  confidencePct: number
  rateSource: 'tier_default' | 'city_override'
  cityLabel: string
  assumptions: string[]
  breakdown: EstimateBreakdownItem[]
}

File: backend/app/schemas/estimator.py
from typing import Literal

from pydantic import BaseModel, Field


CityTier = Literal["metro", "tier1", "tier2", "tier3"]
Quality = Literal["basic", "standard", "premium"]
HouseType = Literal["independent", "villa", "row"]


class EstimateInput(BaseModel):
    plotSqft: float = Field(..., ge=0)
    builtSqft: float = Field(..., ge=0)
    floors: Literal[1, 2, 3, 4]
    houseType: HouseType
    cityTier: CityTier
    citySlug: str | None = None
    quality: Quality
    basement: bool = False
    parking: bool = False
    lift: bool = False
    vastu: bool = False
    solarReady: bool = False


class EstimateBreakdownItem(BaseModel):
    key: str
    label: str
    amount: int
    percentage: int
    color: str


class EstimateResult(BaseModel):
    totalCost: int
    low: int
    high: int
    baseCost: int
    addonCost: int
    costPerSqft: int
    timelineMonths: int
    confidencePct: int
    rateSource: Literal["tier_default", "city_override"]
    cityLabel: str
    assumptions: list[str]
    breakdown: list[EstimateBreakdownItem]

